### ISSUE TITLE
Create collapsible tray for Saving Throws

### ISSUE THREAD

**Issue by MaxPat931** [foundryvtt/dnd5e#3897]
It can still be a bit challenging to select the appropriate tokens for damage application for spells/savings throws with larger numbers of affected tokens, adding a collapsible tray similar to attacks for tokens that roll a save from the chat card's button would be a convenient method of accomplishing this, and cut down on chat card spam if the save roll is routed directly here instead of creating cards. Including a right-click option to select all Fails/Successes, similar to Hits/Misses on attacks, would be helpful as well

![image](https://github.com/user-attachments/assets/15c2d2c8-6950-4111-8c05-3e853f1eeb6e)

**Comment by Kapuzenjoe** [comment]
Big plus here!
After my PCs threw an AoE with saving throw on 24 Twig Blights, the tracking was pure chaos.
Keeping the targets+results of the saving throws bundled in a chat window, which also allows you to pass the targets+results to the damage roll, would be a massive QoL for many players/DMs

**Comment by MaxPat931** [comment]
Yea for 24 tokens this would be pretty time consuming, for what it's worth, here is the most efficient way I've found for applying damage/effects from a feature with a Saving Throw is:
1. Player casts spell, places template
2. Player targets tokens
3. Player rolls damage
4. DM selects affected tokens, rolls saves
5. DM selects Successes by shift+clicking failed save card's portraits (or by deselecting tokens, shift+clicking successful save card's portraits)
6. DM goes to damage card, clicks Selected
7. DM selects 1/2 damage for all successes
8. DM clicks Targeted, applies damage to all
9. (if Effect is present) DM selects Failures by deselecting tokens, shift+clicks failed save card's portraits
10. DM goes to item card, applies effect

[SaveProcess.webm](https://github.com/user-attachments/assets/f425c1a5-38f9-49c2-be3e-6a7a6c24e97e)

**Comment by roth-michael** [comment]
I'd like to throw this into consideration, but rather than adding a tray, perhaps instead making use of the Request-type chat card for the targets of a save. Might require some refinement since templates don't currently auto-target, and so there'd need to be a way of saying "okay, targeting is ready." Could also then sensibly default damage application multiplier button states for the eventual damage card based on the results of the requested saves. In its initial iteration this of course wouldn't cover something like Evasion.

**Comment by Kapuzenjoe** [comment]
After trying to address this with a mod as a workaround (see mod here: https://github.com/Kapuzenjoe/save-tray-5e), I’d like to share a few ideas for a more useful approach (at least in my opinion):

- A tray that only displays saving throws is nice, but by itself it doesn’t add a lot of real QoL.
- It would be much more useful if the tray also listed the caster’s current targets, and provided a button per target to trigger the save roll directly. That way you wouldn’t need to click targets on the canvas anymore - just hit the button. This could also support fast-forwarding, etc.
    - Alternatively (for automation-heavy workflows), it could prompt the target to roll via a query/request.
- Additionally: when the normal “Save Roll” button is clicked, the currently selected token(s) should be added to the save tray automatically. If the button is clicked again for the same token(s), the tray should refresh/update that token’s roll result rather than adding duplicates.
- Next step: when clicking the subsequent Damage Roll, it would be great if the tray’s targets were carried over automatically into the damage chat message as targets, and the system could pre-select the default multiplier (e.g. 0.5 or 0 on a successful save) based on the results - while still allowing the GM to adjust manually.
- Also, it would be helpful to have an option to remove targets from the tray (especially relevant for overtime effects, when you pop-out the message and need to adjust targets accordingly).

With these points, a lot of common use cases could be improved without a huge amount of extra complexity. Save-based spells like Sacred Flame would become comparable to attack rolls in terms of workflow: currently, the player often has to announce they’re “attacking”, and the GM then has to search and  manually click the target(s) on the canvas to run the save. This would streamline that significantly.

A template-based targeting workflow would also be interesting here, but I suspect that’s better handled as a separate issue.

**Comment by thatlonelybugbear** [comment]
Having dealt in a module of mine with the damage tray picking up rolled saves and changing the damage multipliers accordingly, I would say that even this small change is a great quality of life addition and not that difficult to implement.

**Comment by Dewi-Jones** [comment]
I had a whole essay written here, but for the sake of all of us I've shortened it to this. View it as my two cents on the topic.

MaxPat's original proposal makes a lot of sense.
A saving throw message should pick up on any saves made from it, and provide the opportunity to (A) apply effects to all or to only failed saves, and (B) provide the opportunity to select successfull saves or failed saves. I'm in full support of this.

I'm also in support of Bugbear's proposal. To me, this would be the icing on the cake, but it doesn't adress the underlying issue on it's own. (Because it ignores the Effect Application)

Two questions in my mind:

- What should happen if a Token made multiple saving throws from one message?
I think only the latest save should be considered. But there is an argument to be made that *any* success should count as a success, for example someone forgetting they have advantage, keeping their first roll and just rolling the second d20. This is a common thing, especially with users not as familiar with the system and the game itself, and different GM's will take different approaches to this situation.
- How should having tokens targeted alter this workflow?
Maybe a way for the DM to quickly select all targeted tokens? Could be useful for spells that do not target an Area, but multiple singluar creatures like Hold Person (2024) or Psychic Scream (XGE)

**Comment by krbz999** [comment]
> What should happen if a Token made multiple saving throws from one message?

Keep the last one only; there aren't any instances I'm aware of of features or spells requiring multiple saves, so we can assume only one saving throw made is relevant. In the case of forgetting advantage mode, the new roll should be made with the correct advantage mode.

**Comment by Dewi-Jones** [comment]
> Keep the last one only; there aren't any instances I'm aware of of features or spells requiring multiple saves, so we can assume only one saving throw made is relevant. In the case of forgetting advantage mode, the new roll should be made with the correct advantage mode.

I agree. But This is because _I_ DM it that way. You forgot to select advantage? OK, the roll doesn't count.
Other DM's might handle it differently, and if we do this, the system would dictate their way of DMing.

The other way around, you could still delete the old messages, assuming this updates the Original Message.
